records/generator.go

```diff
@@ -150,41 +150,44 @@ func (rg *RecordGenerator) findMaster(masters ...string) (state.State, error) {
 
 	// Check if ZK leader is correct
 	if leader != "" {
-		logging.VeryVerbose.Println("Zookeeper says the leader is: ", leader)
+		logging.VeryVerbose.Println("The leader is: ", leader)
 		ip, port, err := getProto(leader)
 		if err != nil {
 			logging.Error.Println(err)
+		} else {
+			if sj, err = rg.loadWrap(ip, port); err == nil && sj.Leader != "" {
```

Comment:
shouldn't these conditions (`err==nil` and `sj.Leader != ""`) be tested separately, as they are below? the logic is so similar, it's actually begging to be consolidated.